components/brave_federated/config_utils.cc

```diff
@@ -0,0 +1,133 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated/config_utils.h"
+
+#include "brave/components/brave_federated/features.h"
+#include "brave/components/brave_federated/task/model.h"
+
+#include "base/json/json_reader.h"
+#include "base/json/json_value_converter.h"
+
+namespace brave_federated {
+
+LearningServiceConfig::LearningServiceConfig() {
+  reconnect_policy_ = {/*.num_errors_to_ignore = */ 0,
+                       /*.initial_delay_ms = */ 10 * 1000,
+                       /*.multiply_factor =*/2.0,
+                       /*.jitter_factor =*/0.0,
+                       /*.maximum_backoff_ms =*/60 * 10 * 1000,
+                       /*.always_use_initial_delay =*/true};
+  request_task_policy_ = {
+      /*.num_errors_to_ignore = */ 0,
+      /*.initial_delay_ms = */
+      features::GetFederatedLearningUpdateCycleInSeconds() * 1000,
+      /*.multiply_factor =*/2.0,
+      /*.jitter_factor =*/0.0,
+      /*.maximum_backoff_ms =*/16 *
+          features::GetFederatedLearningUpdateCycleInSeconds() * 1000,
+      /*.always_use_initial_delay =*/true};
+  post_results_policy_ = {
+      /*.num_errors_to_ignore = */ 0,
+      /*.initial_delay_ms = */
+      features::GetFederatedLearningUpdateCycleInSeconds() * 1000,
+      /*.multiply_factor =*/2.0,
+      /*.jitter_factor =*/0.0,
+      /*.maximum_backoff_ms =*/16 *
+          features::GetFederatedLearningUpdateCycleInSeconds() * 1000,
+      /*.always_use_initial_delay =*/true};
+  model_spec_ = {/* .num_params = */ 32,
+                 /* .batch_size = */ 32,
+                 /* .learning_rate = */ 0.01,
+                 /* .num_iterations = */ 500,
+                 /* .threshold = */ 0.5};
+}
+
+LearningServiceConfig::LearningServiceConfig(const base::FilePath& path)
```

Comment:
why would this ever read directly from a file?

Reply:
Default behaviour is to read configurations from a json file. If the file for any reason is not there, then uses the default configuration. This is to avoid having to recompile the project for changing these configurations.